### PR TITLE
Add table aws_rds_db_event_subscription closes #608

### LINKED ISSUE #608: Add table aws_rds_db_event_subscription
**References**
https://docs.aws.amazon.com/sdk-for-go/api/service/rds/#EventSubscription

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_rds_db_event_subscription []

PRETEST: tests/aws_rds_db_event_subscription

TEST: tests/aws_rds_db_event_subscription
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_vpc.my_vpc: Creating...
aws_sns_topic.named_test_resource: Creating...
aws_sns_topic.named_test_resource: Creation complete after 4s [id=arn:aws:sns:us-east-1:986325076436:turbottest13921]
aws_vpc.my_vpc: Still creating... [10s elapsed]
aws_vpc.my_vpc: Creation complete after 14s [id=vpc-0d4bf4d313c231049]
aws_internet_gateway.igw: Creating...
aws_internet_gateway.igw: Creation complete after 6s [id=igw-0b987ca7d98d0a26a]
aws_subnet.my_subnet1: Creating...
aws_subnet.my_subnet2: Creating...
aws_subnet.my_subnet2: Creation complete after 3s [id=subnet-0cbfef01df8898fd4]
aws_subnet.my_subnet1: Creation complete after 3s [id=subnet-0e4ba5a8a42d17a02]
aws_db_subnet_group.my_subnetgroup: Creating...
aws_db_subnet_group.my_subnetgroup: Creation complete after 5s [id=turbottest13921]
aws_db_instance.named_test_resource: Creating...
aws_db_instance.named_test_resource: Still creating... [10s elapsed]
aws_db_instance.named_test_resource: Still creating... [20s elapsed]
aws_db_instance.named_test_resource: Still creating... [30s elapsed]
aws_db_instance.named_test_resource: Still creating... [40s elapsed]
aws_db_instance.named_test_resource: Still creating... [50s elapsed]
aws_db_instance.named_test_resource: Still creating... [1m0s elapsed]
aws_db_instance.named_test_resource: Still creating... [1m10s elapsed]
aws_db_instance.named_test_resource: Still creating... [1m20s elapsed]
aws_db_instance.named_test_resource: Still creating... [1m30s elapsed]
aws_db_instance.named_test_resource: Still creating... [1m40s elapsed]
aws_db_instance.named_test_resource: Still creating... [1m50s elapsed]
aws_db_instance.named_test_resource: Still creating... [2m0s elapsed]
aws_db_instance.named_test_resource: Still creating... [2m10s elapsed]
aws_db_instance.named_test_resource: Still creating... [2m20s elapsed]
aws_db_instance.named_test_resource: Still creating... [2m30s elapsed]
aws_db_instance.named_test_resource: Still creating... [2m40s elapsed]
aws_db_instance.named_test_resource: Still creating... [2m50s elapsed]
aws_db_instance.named_test_resource: Still creating... [3m0s elapsed]
aws_db_instance.named_test_resource: Still creating... [3m10s elapsed]
aws_db_instance.named_test_resource: Still creating... [3m20s elapsed]
aws_db_instance.named_test_resource: Still creating... [3m30s elapsed]
aws_db_instance.named_test_resource: Creation complete after 3m37s [id=turbottest13921]
aws_db_event_subscription.named_test_resource: Creating...
aws_db_event_subscription.named_test_resource: Still creating... [10s elapsed]
aws_db_event_subscription.named_test_resource: Still creating... [20s elapsed]
aws_db_event_subscription.named_test_resource: Still creating... [30s elapsed]
aws_db_event_subscription.named_test_resource: Creation complete after 35s [id=turbottest13921]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 8 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
customer_aws_id = 986325076436
region_name = us-east-1
resource_aka = arn:aws:rds:us-east-1:986325076436:es:turbottest13921
resource_id = turbottest13921
resource_name = turbottest13921
sns_topic_arn = arn:aws:sns:us-east-1:986325076436:turbottest13921

Running SQL query: test-get-query.sql
[
  {
    "arn": "arn:aws:rds:us-east-1:986325076436:es:turbottest13921",
    "cust_subscription_id": "turbottest13921",
    "customer_aws_id": "986325076436",
    "enabled": false,
    "sns_topic_arn": "arn:aws:sns:us-east-1:986325076436:turbottest13921",
    "status": "active"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:rds:us-east-1:986325076436:es:turbottest13921",
    "cust_subscription_id": "turbottest13921",
    "enabled": false
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:rds:us-east-1:986325076436:es:turbottest13921"
    ],
    "cust_subscription_id": "turbottest13921",
    "title": "turbottest13921"
  }
]
✔ PASSED

POSTTEST: tests/aws_rds_db_event_subscription

TEARDOWN: tests/aws_rds_db_event_subscription

SUMMARY:

1/1 passed.

```
</details>

# Example query results
<details>
  <summary>Results</summary>

### Basic Info
  
```sql
select
  cust_subscription_id,
  customer_aws_id,
  arn,
  status,
  enabled
from
  aws_rds_db_event_subscription;
```

```
+----------------------+-----------------+----------------------------------------------------------+--------+---------+
| cust_subscription_id | customer_aws_id | arn                                                      | status | enabled |
+----------------------+-----------------+----------------------------------------------------------+--------+---------+
| my-db-subscription   | 986325076436    | arn:aws:rds:us-east-1:986325076436:es:my-db-subscription | active | true    |
+----------------------+-----------------+----------------------------------------------------------+--------+---------+

```

### List DB event subscription which are enable

```sql
select
  cust_subscription_id,
  enabled
from
  aws_rds_db_event_subscription
where
  enabled;
```

```
+----------------------+---------+
| cust_subscription_id | enabled |
+----------------------+---------+
| my-db-subscription   | true    |
+----------------------+---------+

```

</details>

### CODE REVIEW

File: aws-test/tests/aws_rds_db_event_subscription/test-get-query.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+select cust_subscription_id, customer_aws_id, arn, enabled, sns_topic_arn, status
+from
+  aws.aws_rds_db_event_subscription
+where
+  cust_subscription_id = '{{ resourceName }}'
```
Comment (khushboo9024) [inline aws-test/tests/aws_rds_db_event_subscription/test-get-query.sql:5]:
```suggestion
select cust_subscription_id, customer_aws_id, arn, enabled, sns_topic_arn, status
from aws.aws_rds_db_event_subscription
where cust_subscription_id = '{{ resourceName }}'
```

File: aws-test/tests/aws_rds_db_event_subscription/test-list-query.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+select cust_subscription_id, arn, enabled
+from aws.aws_rds_db_event_subscription
+where arn = '{{ output.resource_aka.value }}'
```
Comment (khushboo9024) [inline aws-test/tests/aws_rds_db_event_subscription/test-list-query.sql:3]:
```suggestion
where arn = '{{ output.resource_aka.value }}';
```

File: aws-test/tests/aws_rds_db_event_subscription/test-not-found-query.sql [hunk 3]
```diff
@@ -0,0 +1,3 @@
+select cust_subscription_id, arn, status
+from aws.aws_rds_db_event_subscription
+where cust_subscription_id = 'dummy-{{ resourceName }}'
```
Comment (khushboo9024) [inline aws-test/tests/aws_rds_db_event_subscription/test-not-found-query.sql:3]:
```suggestion
where cust_subscription_id = 'dummy-{{ resourceName }}';
```

File: aws-test/tests/aws_rds_db_event_subscription/test-turbot-query.sql [hunk 4]
```diff
@@ -0,0 +1,3 @@
+select cust_subscription_id, title, akas
+from aws.aws_rds_db_event_subscription
+where cust_subscription_id = '{{ resourceName }}'
```
Comment (khushboo9024) [inline aws-test/tests/aws_rds_db_event_subscription/test-turbot-query.sql:3]:
```suggestion
where cust_subscription_id = '{{ resourceName }}';
```

File: aws-test/tests/aws_rds_db_event_subscription/variable.tf [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+variable "resource_name" {
+  type        = string
+  default     = "turbot-test-20200125-create-update"
+  description = "Name of the resource used throughout the test."
+}
+
+variable "aws_profile" {
+  type        = string
+  default     = "default"
+  # default     = "integration-tests"
```
Comment (khushboo9024) [inline aws-test/tests/aws_rds_db_event_subscription/variable.tf:11]: Remove commented lines.

File: docs/tables/aws_rds_db_event_subscription.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  cust_subscription_id,
+  customer_aws_id,
+  arn,
+  status,
+  enabled
+from
+  aws_rds_db_event_subscription;
+```
+
+### List DB event subscription which are enable
```
Comment (khushboo9024) [inline docs/tables/aws_rds_db_event_subscription.md:20]:
```suggestion
### List enabled DB event subscription
```

File: aws/table_aws_rds_db_event_subscription.go [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+	}
+	return nil, nil
+}
+
+//// TRANSFORM FUNCTION
+
+func convertStringToRFC3339Timestamp(_ context.Context, d *transform.TransformData) (interface{}, error) {
+	data := d.HydrateItem.(*rds.EventSubscription)
+
+	// 2021-08-18 06:24:35
```
Comment (khushboo9024) [inline aws/table_aws_rds_db_event_subscription.go:156]: Remove this commented lines

**Review by khushboo9024 — APPROVED** [review]

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**